### DISCUSSION TITLE
problem with compiling software for IR controller for Daikin air conditioner

### DISCUSSION THREAD

**Discussion by comkubus7** [avbor/HomeAssistantConfig#2] [General]
Hi @avbor, I tried to create a connection between my HomeAssistant and my Daikin air conditioner following your instructions. However, I encountered an issue when compiling the software for the IR controller, which gave me the following error:

`src\daikin_ir.h: In member function 'virtual void DaikinAC::setup()':`
`src\daikin_ir.h:32:82: error: no matching function for call to 'clamp(float&, int, int)'`

I'm struggling with this and I don't know how to fix it. What version of the IRremoteESP8266 library did you use? Have you encountered a similar problem or do you know how to solve it?

**Comment by avbor** [comment]
Hi!
Try to use this file https://github.com/avbor/HomeAssistantConfig/blob/master/esphome/include/daikin_ir_v2.h
and config like this:
```
esphome:
  includes: 
    - include/daikin_ir_v2.h
  libraries:
    - IRremoteESP8266
```

**Comment by comkubus7** [comment]
I see my bad..:-) With change to v2 everything works... :-) By the way, you haven't figured out how to decode the ir signal from the remote? I have the same model and I have to say that only your version somehow works for me... So good job!:-)

**Reply by avbor** [reply]
> With change to v2 everything works

That's great!
Because I still haven't tried it in mine...
My IR still works on a very old version of ESPHome =)

As far as I understand, Daikin has a large number of IR protocols, but only some of them are available in ESPHome. Why this is so - I don't know, my request for support for these models (https://github.com/esphome/issues/issues/1726) has not been answered since 2020.

**Reply by avbor** [reply]
On a side note, I haven't tried the newer versions of ESPHome to control FTXB, maybe their protocol support has been added at some point.
https://esphome.io/components/climate/climate_ir.html
Maybe via Arduino-HeatpumpIR:
![image](https://user-images.githubusercontent.com/32452033/223335672-e8daf8f8-9fb7-4b4a-af22-cdaaa8c57cc3.png)
